presto-hive/src/main/java/com/facebook/presto/hive/HiveMaterializedViewUtils.java

```diff
@@ -0,0 +1,288 @@
+                            .collect(toLinkedMap(
+                                    rightToLeftPredicatesKeyMap::get,
+                                    rightPredicateKeyValue::get)));
+
+            rightPredicatesMappedToLeftCommonKeys.add(rightPredicateMappedToLeftCommonKeys);
+        }
+
+        ImmutableList.Builder<TupleDomain<String>> leftMinusRight = ImmutableList.builder();
+
+        for (TupleDomain<String> leftPredicate : leftPredicatesInfo.getPredicateDisjuncts()) {
+            LinkedHashMap<String, NullableValue> leftPredicateKeyValue = getLinkedHashMap(
+                    extractFixedValues(leftPredicate).orElseThrow(() -> new IllegalStateException("leftPredicateKeyValue is not present!")));
+
+            LinkedHashMap<String, NullableValue> leftPredicateMappedToLeftCommonKeys = getLinkedHashMap(
+                    leftPredicateKeyValue.keySet().stream()
+                            .filter(leftMappedCommonKeys::contains)
+                            .collect(Collectors.toMap(
+                                    columnName -> columnName,
+                                    leftPredicateKeyValue::get)));
+
+            if (!rightPredicatesMappedToLeftCommonKeys.contains(leftPredicateMappedToLeftCommonKeys)) {
+                leftMinusRight.add(leftPredicate);
+            }
+        }
+
+        return new MaterializedDataPredicates(leftMinusRight.build(), leftPredicatesInfo.getColumnNames());
+    }
+
+    public static MaterializedDataPredicates getEmptyMaterializedViewDataPredicates()
+    {
+        return EMPTY_MATERIALIZED_VIEW_DATA_PREDICATES;
+    }
+
+    private static <K, V> LinkedHashMap<K, V> getLinkedHashMap(Map<K, V> map)
+    {
+        return (map instanceof LinkedHashMap) ? (LinkedHashMap<K, V>) map : new LinkedHashMap<>(map);
+    }
+
+    public static Map<SchemaTableName, Map<String, String>> getViewToBasePartitionMap(
+            Table view,
+            List<Table> baseTables,
+            Map<String, Map<SchemaTableName, String>> viewToBaseColumnMap)
+    {
+        List<String> viewPartitions = view.getPartitionColumns()
+                .stream()
+                .map(Column::getName)
+                .collect(Collectors.toList());
+
+        Map<SchemaTableName, List<String>> baseTablePartitions = baseTables.stream()
+                .collect(Collectors.toMap(
+                        table -> new SchemaTableName(table.getDatabaseName(), table.getTableName()),
+                        table -> table.getPartitionColumns().stream().map(Column::getName).collect(toImmutableList())));
+
+        ImmutableMap.Builder<SchemaTableName, Map<String, String>> viewToBasePartitionMap = ImmutableMap.builder();
+        for (SchemaTableName baseTable : baseTablePartitions.keySet()) {
+            Map<String, String> partitionMap = new HashMap<>();
+            for (String viewPartition : viewPartitions) {
+                for (String basePartition : baseTablePartitions.get(baseTable)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.